esp/services/ws_workunits/ws_workunitsService.cpp

```diff
@@ -3179,11 +3190,34 @@ void getWorkunitCluster(IEspContext &context, const char *wuid, SCMStringBuffer
     }
 }
 
-void getFileResults(IEspContext &context, const char *logicalName, const char *cluster, __int64 start, unsigned &count, __int64 &total,
+IDistributedFile *CWsWorkunitsEx::lookupLogicalName(IEspContext &context, const char *logicalName)
+{
+    StringBuffer userID;
+    context.getUserID(userID);
+    Owned<IUserDescriptor> userDesc;
+    if (!userID.isEmpty())
+    {
+        userDesc.setown(createUserDescriptor());
+        userDesc->set(userID, context.queryPassword(), context.querySignature());
+    }
+
```

Comment:
is this is common/reusable pattern in other places?
I think v. similar code is in (e.g) ws_dfuService.cpp  - and perhaps other places?

I think it's ok as is now, but I'd consider commoning up such things if possible sometime (in a different PR).

Reply:
https://track.hpccsystems.com/browse/HPCC-24260